### PR TITLE
fix an error when testing a result from a method that returns a promise

### PR DESCRIPTION
**Pull request by bbaldino**

### CODE REVIEW

File: conference.js [hunk 1]
```diff
@@ -1873,34 +1873,40 @@ export default {
...
-                // Ugly way to synchronize real device IDs with local
-                // storage and settings menu. This is a workaround until
-                // getConstraints() method will be implemented in browsers.
-                if (localAudio) {
-                    APP.settings.setMicDeviceId(
-                        localAudio.getDeviceId(), false);
-                }
+        JitsiMeetJS.mediaDevices.isDeviceListAvailable()
+            .then(() => {
+                if (JitsiMeetJS.mediaDevices.isDeviceChangeAvailable()) {
```
Comment (virtuacoplenny) [inline conference.js]: I wonder if you should use the return value of isDeviceListAvailable.

Comment (virtuacoplenny) [inline conference.js]: Style thought: What about returning early if falsy instead of nesting the logic in the if?

Comment (bbaldino) [inline conference.js]: yes i suppose that would be a good idea :)

File: conference.js [hunk 2]
```diff
@@ -1977,35 +1977,42 @@ export default {
...
-                // storage and settings menu. This is a workaround until
-                // getConstraints() method will be implemented in browsers.
-                if (localAudio) {
-                    APP.settings.setMicDeviceId(
-                        localAudio.getDeviceId(), false);
-                }
+        JitsiMeetJS.mediaDevices.isDeviceListAvailable()
+            .then(isDeviceListAvailable => {
+                if (isDeviceListAvailable
+                        && JitsiMeetJS.mediaDevices.isDeviceChangeAvailable()) {
```
Comment (virtuacoplenny) [inline conference.js]: Style thought: What about returning early if falsy instead of nesting the logic in the if?

**Pull request merged**